lsp-types/src/Language/LSP/Types/CallHierarchy.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+{- | Since LSP 3.16.0 -}
+module Language.LSP.Types.CallHierarchy where
+
+import Data.Aeson.TH
+import Data.Aeson.Types ( Value )
+import Data.Text ( Text )
+
+import Language.LSP.Types.Common
+import Language.LSP.Types.DocumentSymbol
+import Language.LSP.Types.Location
+import Language.LSP.Types.Progress
+import Language.LSP.Types.StaticRegistrationOptions
+import Language.LSP.Types.TextDocument
+import Language.LSP.Types.Uri
+import Language.LSP.Types.Utils
+
+
+data CallHierarchyClientCapabilities =
+  CallHierarchyClientCapabilities
+    { _dynamicRegistration :: Maybe Bool }
+    deriving (Show, Read, Eq)
+deriveJSON lspOptions ''CallHierarchyClientCapabilities
+
+makeExtendingDatatype "CallHierarchyOptions" [''WorkDoneProgressOptions] []
+deriveJSON lspOptions ''CallHierarchyOptions
+
+makeExtendingDatatype "CallHierarchyRegistrartionOptions"
+  [ ''TextDocumentRegistrationOptions
+  , ''CallHierarchyOptions
+  , ''StaticRegistrationOptions
+  ]
+  []
+deriveJSON lspOptions ''CallHierarchyRegistrartionOptions
+
+makeExtendingDatatype "CallHierarchyPrepareParams"
+  [''TextDocumentPositionParams, ''WorkDoneProgressParams] []
+deriveJSON lspOptions ''CallHierarchyPrepareParams
+
+data CallHierarchyItem =
+  CallHierarchyItem
+    { _name :: Text
```

Comment:
Can we include haddock corresponding to the doc in the LSP spec for each item? Tedious, but matching the spec is a simple policy and more field documentation is almost always better.

Reply:
```haskell
-- <https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_prepareCallHierarchy>
data CallHierarchyItem = ...
```

Should it look like this? I have little knowledge about haddock.